packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/scroll/ReactHorizontalScrollContainerView.java

```diff
@@ -50,24 +48,7 @@ protected void onLayout(boolean changed, int left, int top, int right, int botto
       int newLeft = 0;
       int width = right - left;
       int newRight = newLeft + width;
-      setLeft(newLeft);
-      setRight(newRight);
-
-      /**
-       * Note: in RTL mode, *when layout width changes*, we adjust the scroll position. Practically,
-       * this means that on the first (meaningful) layout we will go from position 0 to position
-       * (right - screenWidth). In theory this means if the width of the view ever changes during
-       * layout again, scrolling could jump. Which shouldn't happen in theory, but... if you find a
-       * weird product bug that looks related, keep this in mind.
-       */
-      if (mCurrentWidth != getWidth()) {
-        // Call with the present values in order to re-layout if necessary
-        ReactHorizontalScrollView parent = (ReactHorizontalScrollView) getParent();
-        // Fix the ScrollX position when using RTL language
-        int offsetX = parent.getScrollX() + getWidth() - mCurrentWidth - parent.getWidth();
-        parent.scrollTo(offsetX, parent.getScrollY());
-      }
+      setLeftTopRightBottom(newLeft, top, newRight, bottom);
```

Comment:
We are noticing crashes due to this PR because of this line. [`setLeftTopRightBottom` was added in API level 29 according to the docs](https://developer.android.com/reference/android/view/View#setLeftTopRightBottom(int,%20int,%20int,%20int)) but that is not handled here. We are going to test the workaround of using `setLeft`, `setTop`, `setRight`, and `setBottom` instead.

Reply:
Good catch! This change was made before some protections we have to avoid calling too new APIs. Would accept PR fixing this.

Reply:
I will open one

Reply:
https://github.com/facebook/react-native/pull/44033